File: Tyche.iOS/DataPool/Sources/DataPool/Application/JoinPoolUseCase.swift
public class JoinPoolUseCase {
    private let poolRepository: PoolRepository

    public init(poolRepository: PoolRepository) {
        self.poolRepository = poolRepository
    }

    public func execute(joinPoolInput: JoinPoolInput) async -> Result<Void, Error> {
        await poolRepository.joinPool(joinPoolInput: joinPoolInput)
    }
}

File: Android/data/pool/src/main/java/com/felipearpa/tyche/data/pool/application/JoinPoolUseCase.kt
package com.felipearpa.tyche.data.pool.application

import com.felipearpa.tyche.data.pool.domain.JoinPoolInput
import com.felipearpa.tyche.data.pool.domain.PoolRepository

class JoinPoolUseCase internal constructor(private val poolRepository: PoolRepository) {
    suspend fun execute(joinPoolInput: JoinPoolInput) =
        poolRepository.joinPool(joinPoolInput = joinPoolInput)
}
